### PR TITLE
Reanimate validates_by_schema

### PR DESCRIPTION
**Pull request by codez**
I upgraded the code to work with rails 4.2, added some tests for the options, simplified the test setup (no more rails, just activerecord) and applied the rubocop auto-fixes.

Are you still maintaining this project? It would be great if you could update the code and release a new gem version. If not, I would also be willing to adopt the project. You would just have to give me push rights for rubygems.org.

I know this is not the nicely separated pull request, but let's see if you like the idea of reanimating this library first ;)

### CODE REVIEW

**Comment by joshwlewis** [comment]
Hi! Sorry for the delayed response, I'm not sure that I got notified when this PR came in.

Let me take a closer look at this, but overall I think it looks great.

File: Gemfile.lock [hunk 1]
```diff
@@ -1,121 +1,102 @@
 PATH
   remote: .
   specs:
-    validates_by_schema (0.2.2)
-      rails (>= 3.1.0)
+    codez-validates_by_schema (0.3.1)
```
Comment (joshwlewis) [inline Gemfile.lock]: Can you fix this, or open a new PR where this points to the actual gem?

File: validates_by_schema.gemspec [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
+$LOAD_PATH.push File.expand_path('../lib', __FILE__)
 
 # Maintain your gem's version:
-require "validates_by_schema/version"
+require 'validates_by_schema/version'
 
 # Describe your gem and declare its dependencies:
 Gem::Specification.new do |s|
-  s.name        = "validates_by_schema"
+  s.name        = 'codez-validates_by_schema'
```
Comment (joshwlewis) [inline validates_by_schema.gemspec]: I won't be able to merge this either -- can you also fix this?

File: validates_by_schema.gemspec [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
   s.version     = ValidatesBySchema::VERSION
-  s.authors     = ["Josh Lewis"]
-  s.email       = ["josh.w.lewis@gmail.com"]
-  s.homepage    = "http://github.com/joshwlewis/validates_by_schema"
-  s.summary     = "Automatic validation based on your database schema column types and limits."
-  s.description = "Keep your code DRY by inferring column validations from table properties! Automagically validate presence, length, numericality, and inclusion of ActiveRecord backed columns."
+  s.authors     = ['Josh Lewis', 'Pascal Zumkehr']
+  s.email       = ['josh.w.lewis@gmail.com', 'spam@codez.ch']
+  s.homepage    = 'http://github.com/codez/validates_by_schema'
```
Comment (joshwlewis) [inline validates_by_schema.gemspec]: Again, this isn't valid. Can you point this at github.com/joshwlewis/validates_by_schema? (not sure, might make sense to create a new PR for this stuff)

File: spec/validations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+  context 'plain' do
+    subject { Contraption.new attributes }
+
+    context 'validates columns of type' do
+      context :string do
+        it { should_not validate_presence_of(:name) }
+        it { should validate_length_of(:name).is_at_most(50) }
+        it { should validate_presence_of(:model) }
+        if ENV['DB'] == 'mysql'
```
Comment (joshwlewis) [inline spec/validations_spec.rb]: Why are we skipping pg here?

**Comment by joshwlewis** [comment]
This looks really good. I'd be happy to accept this if you can fix and/or reply to my comments above. 

Out of curiosity, are you using this (or your fork) in a production app?

**Comment by codez** [comment]
Replaced by #2

**Pull request closed without merging**